app/models/labware_creators/merged_plate.rb

```diff
@@ -75,7 +75,7 @@ def request_hash(source_well, child_plate, additional_parameters)
 
     # validation to check the number of barcodes scanned matches the number of expected purposes from the configuration
     def all_source_barcodes_entered?
-      return if minimal_barcodes.size == expected_source_purposes.size
+      return false if minimal_barcodes.size == expected_source_purposes.size
```

Comment:
I've realised what's bothering me about these changes! 

I think that the return value is not actually used from these validation methods. The key thing is that if there is an `error` added, it causes validation to fail.

Can someone confirm that's correct? @stevieing @andrewsparkes 

If that's correct, then I think these changes are misleading. Returning `false` reads to me like the validation has failed, whereas it's actually succeeded. I think the correct solution is that these methods should not end in a question mark i.e. there should not be an expectation to return a boolean value.

Reply:
(agreed in Slack)